scripts/mailing_lists/migrate_project_mailing_lists.py

```diff
@@ -0,0 +1,173 @@
+# -*- coding: utf-8 -*-
+"""Create mailing lists for all top-level projects
+"""
+import logging
+import sys
+
+from modularodm import Q
+
+from framework.mongo import database
+from framework.mongo.utils import paginated
+from framework.transactions.context import TokuTransaction
+
+from website.app import init_app
+from website.mailing_list.utils import create_list as real_create_list
+from website.models import Node
+from website.notifications.model import NotificationSubscription
+from website.notifications.utils import to_subscription_key
+from scripts import utils as script_utils
+
+logger = logging.getLogger(__name__)
+
+EVENT = 'mailing_list_events'
+SUBSCRIPTION_TYPE = 'email_transactional'
+
+def fake(*args, **kwargs):
+    # Override actual method to prevent outgoing calls
+    return
+
+def get_targets():
+    return paginated(Node, Q('mailing_enabled', 'eq', None))
+
+def migrate(dry_run=True):
+    if dry_run:
+        create_list = fake
+    else:
+        create_list = real_create_list
+
+    successful_enables = []
+    successful_noops = []
+    successful_disables = []
+    mailgun_failures = []
+    unknown_failures = {}
+    nodes = get_targets()
+    ncount = Node.find().count()
+    i = 0
+    logger.info('Preparing to migrate {} nodes.'.format(ncount))
+    for node in nodes:
+        i += 1
+        try:
+            assert isinstance(node, Node)
+        except:
+            continue
+
+        if not node.is_mutable_project:
+            try:
+                node_kind = 'immutable project'
+                if node.is_registration:
+                    node_kind = 'registration'
+                elif node.is_collection:
+                    node_kind = 'collection' 
+                elif node.is_deleted:
+                    node_kind = 'deleted project' 
+                logger.info('({done}/{total}) Disabling mailing list for {kind} {_id}'.format(done=i, total=ncount, _id=node._id, kind=node_kind)
+                )
+                database['node'].find_and_modify(
+                    {'_id': node._id},
+                    {'$set': {'mailing_enabled': False,
+                              'mailing_updated': False}
+                    }
+                )
+                successful_disables.append(node._id)
+            except Exception as e:
+                logger.exception('Error while handling node {}'.format(node._id))
+                unknown_failures[node._id] = e
+        else:
+            subscription = NotificationSubscription.load(to_subscription_key(node._id, EVENT))
+            if subscription:
+                assert node.mailing_enabled
+                logger.info('({0}/{1}) Mailing list for node {2} already enabled'.format(i, ncount, node._id))
+                successful_noops.append(node._id)
+                continue
```

Comment:
What if the script is canceled or fails in the middle of subscribing contributors? Wouldn't we be bailing out prematurely here?